ipxe/dhcp_options.go

```diff
@@ -68,16 +68,16 @@ func Setup(rep *dhcp4.Packet) {
 	rep.SetIP(dhcp4.OptionLogServer, conf.PublicSyslogIPv4) // Have iPXE send syslog to me.
 }
 
-var packetVersion = []byte{1, 0, 255}
+var ouriPXEVersion = []byte{1, 0, 255}
```

Comment:
what does this correspond to? version 255? how is that our ipxe version?

Reply:
We override the iPXE version that iPXE uses for itself and sends to the dhcp server. This is how we distinguish an iPXE burned into a nic by a vendor vs the one we build and send via tftp. There are better options here, but not as convenient to apply. We can define the IPXE_VERSION via an env var when building ipxe but other options would require a patch to one or more locations in the iPXE code. This version number is no longer incrementing in upstream iPXE so is low-risk to modify.

Reply:
I believe this should be a code comment. This is tribal knowledge that needs documenting somewhere.

Reply:
yep will be ther

Reply:
also in the commit message ;)

Reply:
done, see: [`e396a1d` (#149)](https://github.com/tinkerbell/boots/pull/149/commits/e396a1d3cafe8b8214389241df7abaed84d98295)

Reply:
woops well I got confused.

Reply:
sigh this was well documented in the commit that introduced this change but was wiped away when open sourcing boots and history was reset... :(. I'll fix that now.

Reply:
ok this too has been commented on in code and expanded on in the commit message.

Reply:
thanks for the update. Would it be possible to add a direct pointer to where in Boots we are overriding this version? This is helpful but still a bit opaque.

Reply:
Yep, good call will add to comment.

Reply:
Updated the commit, for ref its here: https://github.com/tinkerbell/boots/blob/main/ipxe/ipxe/build.sh#L43